### PR TITLE
gnutls: Fix CVE-2024-28834, CVE-2024-28835

### PR DESCRIPTION
**Pull request by dmcilvaney**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [ ] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [ ] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [ ] All package sources are available
- [ ] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [ ] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [ ] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ ] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

Grab patches from 3.8.4 for both CVE-2024-28834 and CVE-2024-28835: https://gitlab.com/gnutls/gnutls/-/merge_requests/1823/commits?commit_id=e369e67a62f44561d417cb233acc566cc696d82d

###### Change Log  

- Patch gnutls CVE-2024-28834
- Patch gnutls CVE-2024-28835
- 
###### Does this affect the toolchain?  

**NO**

###### Links to CVEs  
- https://brave-ocean-0baeae310.5.azurestaticapps.net/#/cve/CVE-2024-28834
- https://brave-ocean-0baeae310.5.azurestaticapps.net/#/cve/CVE-2024-28835

###### Test Methodology

- Local
- ~https://dev.azure.com/mariner-org/mariner/_build/results?buildId=666619&view=results~
- https://dev.azure.com/mariner-org/mariner/_build/results?buildId=667229&view=results

### CODE REVIEW

**Comment by rikenm1** [comment]
there's PR check failure. looks like some of the test failed

**Comment by dmcilvaney** [comment]
- ah, user error on the buddybuild, queued for an extended build not a core build
- fasttrack tests (checked against https://msit.powerbi.com/groups/68ae8877-10a3-4b3f-b8ad-9c9cb49f9519/reports/6ce33f6d-1884-4aa8-940c-9120acde2746/ReportSectionc68abe8394b89057c4cc?experience=power-bi)
  - all are known failures

File: SPECS/gnutls/gnutls.spec [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
-Release:        1%{?dist}
+Release:        2%{?dist}
 License:        GPLv3+ AND LGPLv2.1+
 Vendor:         Microsoft Corporation
 Distribution:   Azure Linux
 Group:          System Environment/Libraries
 URL:            https://www.gnutls.org
 Source0:        https://www.gnupg.org/ftp/gcrypt/gnutls/v3.8/%{name}-%{version}.tar.xz
+# Patch taken from 3.8.4 release
+Patch:          CVE-2024-28834.patch
```
Comment (anphel31) [inline SPECS/gnutls/gnutls.spec:12]: Why not Patch1, Patch2 ?

Comment (dmcilvaney) [reply SPECS/gnutls/gnutls.spec:12]: I recall reading that un-numbered patches were generally preferred now upstream, but I can't for the life of me find it again... easy to just add the explicit numbers back (but it does work without).

File: SPECS/gnutls/gnutls.spec [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
+# Patch taken from 3.8.4 release
+Patch:          CVE-2024-28835.patch
```
Comment (dmcilvaney) [inline SPECS/gnutls/gnutls.spec:14]:
```suggestion
Patch1:          CVE-2024-28834.patch
# Patch taken from 3.8.4 release
Patch2:          CVE-2024-28835.patch
```

File: SPECS/gnutls/gnutls.spec (hunk 1 above)
Comment (dmcilvaney) [reply SPECS/gnutls/gnutls.spec]: Fixed

**Review by jslobodzian — APPROVED** [review]

**Review by anphel31 — APPROVED** [review]

**Pull request merged**